opencensus/trace/span.h

```diff
@@ -163,15 +166,19 @@ class Span final {
   // Returns span_impl_, only used for testing.
   std::shared_ptr<SpanImpl> span_impl_for_test() { return span_impl_; }
 
+  // Swaps contents, used for Context.
+  friend void swap(Span& a, Span& b);
+
   // Spans that aren't sampled still have a valid SpanContext that propagates,
   // but no span_impl_.
-  const SpanContext context_;
+  SpanContext context_;
 
   // Shared pointer to the underlying Span representation. This is nullptr for
   // Spans which are not recording events. This is an implementation detail, not
   // part of the public API.
-  const std::shared_ptr<SpanImpl> span_impl_;
+  std::shared_ptr<SpanImpl> span_impl_;
```

Comment:
We made the shared_ptr non-const (and also the SpanContext above) so that we can use `swap()` (L170) and `operator=` (#191) to modify a Span object in-place, so that Context can have a Span member (as opposed to pointer to Span).

Also in (#191) we marked all Span methods const because:
1. They don't act on the shared_ptr, they act on the pointee.
2. In future we can replace the shared_ptr with something like opencensus-java's approach.